src/main/java/it/gov/pagopa/pu/worker/config/DataSourceConfiguration.java

```diff
@@ -0,0 +1,52 @@
+package it.gov.pagopa.pu.worker.config;
+
+import jakarta.persistence.EntityManagerFactory;
+import org.hibernate.boot.model.naming.CamelCaseToUnderscoresNamingStrategy;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.boot.orm.jpa.EntityManagerFactoryBuilder;
+import org.springframework.boot.orm.jpa.hibernate.SpringImplicitNamingStrategy;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.sql.DataSource;
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+@EnableJpaRepositories
+@EnableTransactionManagement
+public class DataSourceConfiguration {
+
+  @Bean
+  @ConfigurationProperties("spring.data.mypay")
+  public DataSource myPayDataSource() {
+    return DataSourceBuilder.create().build();
+  }
```

Comment:
you are using custom properties to configure the DataSource...
It was not a requirement, but OK

let's replace reference to mypay into pu
```suggestion
  @Bean
  @ConfigurationProperties("spring.data.pu")
  public DataSource puDataSource() {
    return DataSourceBuilder.create().build();
  }
```